### PR TITLE
#155 - add wheemove provider

### LINKED ISSUE #155: Add Wheemove provider
:es: Mallorcan provider has multiple cities on Mallorca and Menorca: https://wheemove.com/

image

### PR DESCRIPTION
**Pull request by JakubKermes**
Added Whee Move provider, Mapbox shows cities at wrong places e.g. USA.
It should close #155 and broaden the scope of #86 issue to fix also that.

### CODE REVIEW

File: app/Importers/WheeMoveDataImporter.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            $this->stopExecution = true;
+        }
+
+        return $this;
+    }
+
+    public function transform(): void
+    {
+        $existingCityProviders = [];
+        $this->extract();
```
Comment (vojcc) [inline app/Importers/WheeMoveDataImporter.php:40]: Why calling `extract()` second time?

File: app/Importers/WheeMoveDataImporter.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        $names_temp = $this->sections->first()->filter('span[class="elementor-icon-list-text"]');
+
+        foreach ($names_temp as $name) {
+            $cityNames[] = $name->nodeValue;
+        }
+        $names_temp = $this->sections->last()->filter('span[class="elementor-icon-list-text"]');
+
+        foreach ($names_temp as $name) {
+            $cityNames[] = $name->nodeValue;
+        }
```
Comment (vojcc) [inline app/Importers/WheeMoveDataImporter.php:56]:
```suggestion
        $cityNames = [];
        
        $firstSectionNames = $this->sections->first()->filter('span[class="elementor-icon-list-text"]');
        $lastSectionNames = $this->sections->last()->filter('span[class="elementor-icon-list-text"]');

        foreach ([$firstSectionNames, $lastSectionNames] as $sectionNames) {
            foreach ($sectionNames as $name) {
                $cityNames[] = $name->nodeValue;
            }
        }
```

File: app/Importers/WheeMoveDataImporter.php (hunk 1 above)
Comment (JakubKermes) [reply app/Importers/WheeMoveDataImporter.php:40]: Sorry, it was used temporary for testing. I'll remove it

File: app/Importers/WheeMoveDataImporter.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        foreach ($cityNames as $name) {
+            $provider = $this->load($name, "Spain");
```
Comment (vojcc) [inline app/Importers/WheeMoveDataImporter.php:59]: Assign country name to constant. Check other importers (Hulaj, BitMobility) for reference.

File: public/providers/wheemove.png
Comment (vojcc) [inline public/providers/wheemove.png:1]: Logo image should be 150x100.

File: config/telescope.php
Comment (vojcc) [inline config/telescope.php:1]: I think that config file is not unnecessary.

File: public/providers/wheemove.png
Comment (JakubKermes) [reply public/providers/wheemove.png:1]: Fixed

File: app/Importers/WheeMoveDataImporter.php (hunk 3 above)
Comment (JakubKermes) [reply app/Importers/WheeMoveDataImporter.php]: Fixed

File: config/telescope.php
Comment (JakubKermes) [reply config/telescope.php:1]: It was obviously mistake, already fixed

**Review by vojcc — APPROVED** [review]

**Review by EwelinaSkrzypacz — APPROVED** [review]
For me it works, when I run `php artisan migrate:fresh --seed` and run importers. Please in issue #86 add comment about Wheemove

**Review by AleksandraKozubal — APPROVED** [review]

**Pull request merged**